sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Infrastructure/StorageScope.cs

```diff
@@ -139,6 +139,45 @@ public static async Task<StorageProperties> CreateStorageAccountAsync()
             }
         }
 
+        /// <summary>
+        ///   It creates and returns an account name parsing it from a connection string.
+        /// </summary>
+        ///
+        /// <param name="connectionString">The connection string from the existing Azure storage account.</param>
+        ///
+        /// <returns>The <see cref="StorageProperties" /> that will be used in a given test run.</returns>
+        ///
+        public static StorageProperties CreateStorageAccountFromConnectionString(string connectionString)
+        {
+            if (IsConnectionStringValid(connectionString, out string name))
+            {
+                return new StorageProperties(name, connectionString);
+            }
+
+            throw new ArgumentException("An account name could not be found in the passed connection string");
+        }
+
+        /// <summary>
+        ///   It checks if the connection string contains a valid Azure storage account name.
+        /// </summary>
+        ///
+        /// <param name="connectionString">The connection string.</param>
+        /// <param name="name">The Azure storage account name taken from the connection string.</param>
+        ///
+        /// <returns>
+        ///   <c>true</c> if the connection string is a valid connection string.
+        ///   <c>false</c> if the connection string is not a valid connection string.
+        /// </returns>
+        ///
+        private static bool IsConnectionStringValid(string connectionString, out string name)
+        {
+            var properties = ConnectionStringParser.Parse(connectionString);
```

Comment:
I started coding and out of no where I ended up creating a general purpose connection string parser 😄 

I did it because I wanted to reuse the existing ```Endpoint``` token parser for the following tokens:
- ```SharedAccessKeyName```
- ```SharedAccessKey```

The problem I was facing is that using the existing code wrote for the ```Endpoint``` token relies on ```IndexOf``` in order to find a token, so the ```SharedAccessKey``` token could not be parsed as ```SharedAccessKeyName``` was in its place.

We ultimately did not need to parse ```SharedAccessKeyName``` and ```SharedAccessKey``` as we could have used the original ```ConnectionStringParser``` for that.

However I ended up doing it and I am quite curious to know what your thoughts are about it.